docs/hooks/hook_civicrm_alterReportVar.md

```diff
@@ -98,3 +109,111 @@ columns appropriately.
+          $var['\CRM_Report_OutputHandler_Csv'] = '\My\Namespace\CsvOutputHandler';
+          break;
+
+        case 'actions':
+          // Add my own
+          $var['report_instance.sample'] = ['title' => 'Export Sample'];
+
+          // Don't let people use the built-in pdf output
+          unset($var['report_instance.pdf']);
+          break;
+      }
+    }
+
+    /**
+     * You may also want to implement hook_civicrm_links to update the
+     * links on the report instance listing pages.
+     *
+     * Implementation of hook_civicrm_links.
+     */
+    function example_civicrm_links($op, $objectName, $objectId, &$links, &$mask, &$values) {
+      if ($op == 'view.report.links') {
+        // These have slightly different array keys than other links.
+        // See CRM/Report/Page/InstanceList.php
+        $links['sample'] = [
+          'label' => E::ts('Export to Sample'),
+          'url' => CRM_Utils_System::url(
+            "civicrm/report/instance/{$objectId}",
+            'reset=1&force=1&output=sample'
+          ),
+        ];
+      }
+    }
+
+    <?php
+    /// FILE: My/Namespace/CsvOutputHandler.php
+    ///
+    /// Note we're extending the original built-in CSV class in order to
+    /// override part of its functionality.
+
+    namespace My\Namespace;
+    use CRM_Example_ExtensionUtil as E;
+    class CsvOutputHandler extends \CRM_Report_OutputHandler_Csv {
+      // We just want to override the email body text because here we don't
+      // want to include the link to the report and the stock wording,
+      // otherwise this class works the same as the core one, so this is all
+      // we need to do.
+      public function getMailBody():string {
+        return $this->getForm()->getReportHeader()
+          . E::ts('Here is my custom email body text.')
+          . $this->getForm()->getReportFooter();
+      }
+    }
+
+    <?php
+    /// FILE: My/Namespace/SampleOutputHandler.php
+    ///
+    /// Here we're adding a whole new type of output format.
+
+    namespace My\Namespace;
+    use CRM_Example_ExtensionUtil as E;
+    class SampleOutputHandler extends \Civi\Report\OutputHandlerBase {
+      /**
+       * Override functions from the base class as needed.
+       *
+       * If you really want to start from scratch you don't have to extend,
+       * but then the class declaration needs to implement
+       * \Civi\Report\OutputHandlerInterface.
+       */
+
+      public function isOutputHandlerFor(\CRM_Report_Form $form):bool {
+        // You can have more complicated decision-making, e.g. based on the
+        // report classname, or the instance id which you can get from
+        // $form->getID(), but it should also be related somehow to what
+        // you used in the $varType == 'actions' hook.
+        return ($form->getOutputMode() === 'sample');
+      }
+
+      // override other functions based on what this format is ...
+    }
+
+    // There is also a working sample extension at
```

Comment:
The sample link should be outside the code block and a [text-based link](https://example.org) rather than a https://naked.link.example.org :-)